packages/embark-accounts-manager/src/index.ts

```diff
@@ -0,0 +1,117 @@
+import async from "async";
+import {Embark, Events, Logger} /* supplied by @types/embark in packages/embark-typings */ from "embark";
+import {__} from "embark-i18n";
+import {AccountParser, dappPath} from "embark-utils";
+import Web3 from "web3";
+
+import fundAccount from "./fundAccount";
+
+export default class AccountsManager {
+  private readonly logger: Logger;
+  private readonly events: Events;
+  private accounts: any[];
+  private nodeAccounts: string[];
+  private web3: any;
+  private ready: boolean;
+
+  constructor(private readonly embark: Embark, _options: any) {
+    this.logger = embark.logger;
+    this.events = embark.events;
+    this.accounts = [];
+    this.nodeAccounts = [];
+    this.ready = false;
+
+    this.events.setCommandHandler("accounts-manager:onReady", (cb) => {
+      if (this.ready) {
+        return cb();
+      }
+      this.events.once("accounts-manager:ready", cb);
+    });
+    this.events.request("proxy:onReady", () => {
+      this.parseAccounts();
+    });
+
+    this.events.setCommandHandler("accounts:get", (cb: any) => {
+      cb(null, this.accounts);
+    });
+  }
+
+  private async checkBlockchainRequest(params: any, callback: (error: any, result: any) => void) {
+    if (params.reqData.method === "eth_sendTransaction" && this.accounts.length) {
+      // Check if we have that account in our wallet
+      const account = this.accounts.find((acc) => Web3.utils.toChecksumAddress(acc.address) === Web3.utils.toChecksumAddress(params.reqData.params[0].from));
+      if (account && account.privateKey) {
+        return this.web3.eth.accounts.signTransaction(params.reqData.params[0], account.privateKey , (err: any, result: any) => {
+          if (err) {
+            return callback(err, null);
+          }
+          params.reqData.method = "eth_sendRawTransaction";
+          params.reqData.params = [result.rawTransaction];
+          callback(err, params);
+        });
+      }
+    }
+    callback(null, params);
+  }
+
+  public arrayEqual(arrayA: string[], arrayB: string[]) {
+    if (arrayA.length !== arrayB.length) {
+      return false;
+    } else {
+      return arrayA.every((address, index) => Web3.utils.toChecksumAddress(address) === Web3.utils.toChecksumAddress(arrayB[index]));
+    }
+  }
+
+  private async checkBlockchainResponse(params: any, callback: (error: any, result: any) => void) {
+    if ((params.reqData.method === "eth_accounts" || params.reqData.method === "personal_listAccounts") && this.accounts.length) {
+      if (!this.arrayEqual(params.respData.result, this.nodeAccounts)) {
+        this.nodeAccounts = params.respData.result;
+        this.accounts = AccountParser.parseAccountsConfig(this.embark.config.blockchainConfig.accounts, this.web3, dappPath(), this.logger, this.nodeAccounts);
```

Comment:
It's not really called every time. It's actually usually only called once, but it's there in case the nodeAccounts changes (eg: when someone manually adds an account to the node).
See line 67 to see the condition